From b8ff25eb43c9fa5eb534495a501c73169649d0a6 Mon Sep 17 00:00:00 2001
From: Jiří Klimeš <jklimes@redhat.com>
Date: Tue, 23 Nov 2010 18:06:58 +0000
Subject: core: correct TIMESTAMP type; fixes potential crash when time() is 32bit

The Crash appeared on some platforms in glib while processing variable
arguments.
---
diff --git a/src/system-settings/nm-default-wired-connection.c b/src/system-settings/nm-default-wired-connection.c
index 1cda3d9..0d19dea 100644
--- a/src/system-settings/nm-default-wired-connection.c
+++ b/src/system-settings/nm-default-wired-connection.c
@@ -163,7 +163,7 @@ constructor (GType type,
 		      NM_SETTING_CONNECTION_AUTOCONNECT, TRUE,
 		      NM_SETTING_CONNECTION_UUID, uuid,
 		      NM_SETTING_CONNECTION_READ_ONLY, priv->read_only,
-		      NM_SETTING_CONNECTION_TIMESTAMP, time (NULL),
+		      NM_SETTING_CONNECTION_TIMESTAMP, (guint64) time (NULL),
 		      NULL);
 
 	g_free (id);
--
cgit v0.8.3-6-g21f6
